### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by Sergo007**
Added remove, add methods for config

### CODE REVIEW

File: flow/service.go [hunk 1]
```diff
@@ -4,43 +4,101 @@ import (
...
-func (f *service) FlowPut(conf flowPutRequest) (bool, error) {
+func (f *service) FlowDelete(conf flowDeleteRequest) (bool, error) {
+
+	if conf.Id == "" {
+		return false, errors.New("id is required")
+	}
+	if !strings.HasPrefix(conf.Id, "/templates/") {
+		return false, errors.New("id should start with '/template/'")
+	}
+	ctxForEtcd, _ := context.WithTimeout(context.Background(), 10*time.Second)
```
Comment (antonputra) [inline flow/service.go]: The second return value is error?
```suggestion
	ctxForEtcd, _ := context.WithTimeout(context.Background(), 10*time.Second)
```

File: flow/service.go [hunk 2]
```diff
@@ -4,43 +4,101 @@ import (
...
+	gr, err := f.etcdClientApi.Delete(ctxForEtcd, conf.Id, clientv3.WithPrefix())
+	if err != nil {
+		return false, err
```
Comment (antonputra) [inline flow/service.go]: should we use "errors" package and wrap it in?

File: flow/service.go [hunk 3]
```diff
@@ -4,43 +4,101 @@ import (
...
+		return false, errors.New("id is required")
 	}
-	f.etcdDataBaseClientApi.Put(ctxForEtcd, conf.Id, conf.Config)
+	if !strings.HasPrefix(conf.Id, "/") {
+		return false, errors.New("id should start with '/'")
+	}
+	ctxForEtcd, _ := context.WithTimeout(context.Background(), 10*time.Second)
+	gr, err := f.etcdClientApi.Get(ctxForEtcd, conf.Id)
+	if err != nil {
+		return false, err
```
Comment (antonputra) [inline flow/service.go]: should we use "errors" package and wrap it in?

File: flow/transport.go [hunk 4]
```diff
@@ -13,45 +11,95 @@ import (
 	kithttp "github.com/go-kit/kit/transport/http"
 )
 
+// flowDelete
+type flowDeleteRequest struct {
```
Comment (antonputra) [inline flow/transport.go]: good practice to leave comments for all public fields including "Id"

Comment (antonputra) [reply flow/transport.go]: should it be public?

File: flow/transport.go [hunk 5]
```diff
@@ -13,45 +11,95 @@ import (
...
+}
+
+type flowDeleteResponse struct {
+	Success bool `json:"success"`
+}
+
+func makeFlowDeleteEndpoint(svc Service) endpoint.Endpoint {
+	return func(_ context.Context, request interface{}) (interface{}, error) {
+		req := request.(flowDeleteRequest)
+		Success, err := svc.FlowDelete(req)
```
Comment (antonputra) [inline flow/transport.go]: "Success"?

File: flow/transport.go [hunk 6]
```diff
@@ -13,45 +11,95 @@ import (
...
+			return nil, err
+		}
+		return flowDeleteResponse{Success}, nil
+	}
+}
+
+func decodeFlowDeleteRequest(_ context.Context, r *http.Request) (interface{}, error) {
+	var request flowDeleteRequest
+	if err := json.NewDecoder(r.Body).Decode(&request); err != nil {
+		return nil, err
```
Comment (antonputra) [inline flow/transport.go]: wrap it

File: flow/transport.go [hunk 7]
```diff
@@ -13,45 +11,95 @@ import (
...
+
+func decodeFlowPostRequest(_ context.Context, r *http.Request) (interface{}, error) {
+	var request flowPostRequest
+	if err := json.NewDecoder(r.Body).Decode(&request); err != nil {
+		return nil, err
+	}
+	return request, nil
+}
+
+// flowPut
```
Comment (antonputra) [inline flow/transport.go]: do they have to be public?

File: flow/transport.go [hunk 8]
```diff
@@ -13,45 +11,95 @@ import (
...
-	Success bool   `json:"success"`
-	Err     string `json:"err,omitempty"`
+	Success bool `json:"success"`
 }
 
 func makeFlowPutEndpoint(svc Service) endpoint.Endpoint {
 	return func(_ context.Context, request interface{}) (interface{}, error) {
-		// todo 3) тут мы возвращаем конечную точку достуа а именно принимаем раскодированый json и вызываем метод нашей бизнес логики
 		req := request.(flowPutRequest)
 		Success, err := svc.FlowPut(req)
```
Comment (antonputra) [inline flow/transport.go]: capital S?

File: Gopkg.toml [hunk 9]
```diff
@@ -27,20 +27,20 @@
 
 [[constraint]]
   name = "github.com/coreos/etcd"
-  version = "3.3.10"
```
Comment (antonputra) [inline Gopkg.toml]: why do we need this?

File: flow/service.go [hunk 10]
```diff
@@ -4,43 +4,101 @@ import (
 	"context"
 	"errors"
 	"github.com/coreos/etcd/clientv3"
+	"strings"
 	"time"
 )
 
 // Service это интерфейс нашего сервиса Flow
 type Service interface {
```
Comment (antonputra) [inline flow/service.go]: comment public methods pls

File: flow/service.go [hunk 11]
```diff
@@ -4,43 +4,101 @@ import (
...
+	if conf.Id == "" {
+		return false, errors.New("id is required")
+	}
+	if !strings.HasPrefix(conf.Id, "/") {
+		return false, errors.New("id should start with '/'")
+	}
 	ctxForEtcd, _ := context.WithTimeout(context.Background(), 10*time.Second)
+	gr, err := f.etcdClientApi.Get(ctxForEtcd, conf.Id)
+	if err != nil {
+		return false, err
```
Comment (antonputra) [inline flow/service.go]: wrap an error pls, to provide more context to the caller

File: flow/service.go (hunk 3 above)
Comment (antonputra) [reply flow/service.go]: wrap an error pls, to provide more context to the caller

File: flow/service.go [hunk 12]
```diff
@@ -4,43 +4,101 @@ import (
...
 
 func (f *service) FlowGet(id string) (string, error) {
+	if id == "" {
+		return "", errors.New("id is required")
+	}
+	if !strings.HasPrefix(id, "/") {
+		return "", errors.New("id should start with '/'")
+	}
+
 	ctxForEtcd, _ := context.WithTimeout(context.Background(), 10*time.Second)
```
Comment (antonputra) [inline flow/service.go]: "-" if it is an error why it is not handled properly?

File: flow/transport.go [hunk 13]
```diff
@@ -13,45 +11,95 @@ import (
...
+	var request flowDeleteRequest
+	if err := json.NewDecoder(r.Body).Decode(&request); err != nil {
+		return nil, err
+	}
+	return request, nil
+}
+
+// flowPost
+type flowPostRequest struct {
+	Id     string `json:"id"`
```
Comment (antonputra) [inline flow/transport.go]: comments pls

File: flow/transport.go [hunk 14]
```diff
@@ -13,45 +11,95 @@ import (
...
+	Config string `json:"config"`
+}
+
+type flowPostResponse struct {
+	Success bool `json:"success"`
```
Comment (antonputra) [inline flow/transport.go]: comments pls

File: Gopkg.toml (hunk 9 above)
Comment (Sergo007) [reply Gopkg.toml]: @tnotb 

Why should we use '^'? if we download the latest version, then compatibility with the code will be broken, as for example in java it often happens

**Pull request merged**